cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -430,12 +448,20 @@ extends ColumnStore with CassandraChunkSource with StrictLogging {
     }
   }
 
-  def writePartKeys(ref: DatasetRef,
-                    shard: Int,
-                    partKeys: Observable[PartKeyRecord],
-                    diskTTLSeconds: Long, updateHour: Long,
+  def scanPartKeysByShardKey(ref: DatasetRef, shard: Int, shardKey: Array[Byte]): Observable[Array[Byte]] = {
+    val table = getOrCreateShardKeyToPartKeyTable(ref, shard)
+    table.scanPartKeys(shardKey)
+    // TODO(a_theimer): figure out if token ranges apply here
```

Comment:
These will only apply if we add extra values to the PK to prevent hot-spotting, right?